pkg/oceanbase/operation/server.go

```diff
@@ -21,12 +21,17 @@ import (
 )
 
 func (m *OceanbaseOperationManager) GetServer(s *model.ServerInfo) (*model.OBServer, error) {
-	observer := &model.OBServer{}
-	err := m.QueryRow(observer, sql.GetServer, s.Ip, s.Port)
+	observers, err := m.ListServers()
 	if err != nil {
-		return nil, errors.Wrap(err, "Get observer failed")
+		return nil, err
 	}
-	return observer, nil
+	for _, observer := range observers {
+		if s.Ip == observer.Ip && s.Port == observer.Port {
+			return &observer, nil
```

Comment:
Why do we list all servers here instead query exactly?

Reply:
If query the exactly record, it will return an error when the record is not exits, so I make it return nil, nil instead of return an error for record not found scenario.

Reply:
It would be better that we add `where ip = ? and port = ?` filter when listing observer, we can get empty slice if no servers matching the condition without return an error. Listing all servers and loop them may become an overhead perhaps.